### PR TITLE
Fix test assertions in TestDAP_stepInTargets.py

### PR DESCRIPTION
**Pull request by kendalharland**
The strings this test is using seem to consistently fail to match against the expected values when built & run targeting Windows amd64. This PR updates them to the expected values.

To fix the test and avoid over-specifying for a specific platform, use `assertIn(<target-substring>,...)` to see if we've got the correct target label instead of comparing the demangler output for an exact string match.

### CODE REVIEW

**Comment by kendalharland** [comment]
This PR was really just the result of me looking at the test output on my own machine and copy-pasting the values that were generated by the test's data. I am not familiar with this test's implementation and would appreciate any guidance on whether there really is a bug here, and how I can test this on different platforms/architectures the same way that the llvm-project CI does it.

**Review by jeffreytan81 — CHANGES_REQUESTED** [review]
Have you tried to run this test on Linux? It is passing on my machine. I am pretty sure this might be Windows platform (like compiler and disassembler) specific behavior which you shouldn't change the default value for the test. You either have to disable this test for Windows (which I never tested) or write different expected value for different platform

**Comment by bulbazord** [comment]
One option might be to change the test to match against some list of expected strings. I'm not super familiar with this test but as Jeffrey said this could be OS/Platform dependent.

**Comment by labath** [comment]
I'm not sure if these names come from the demangler or debug info, but windows has different implementations for both, so it not surprising they're different. Assuming we don't care about  the precise formatting of the names, we could just check whether the name of the function is `in` the string (and change the name to something more unique): `assertIn("my_first_step_target", step_in_targets[0]["label"])`

**Comment by kendalharland** [comment]
> I'm not sure if these names come from the demangler or debug info, but windows has different implementations for both, so it not surprising they're different. Assuming we don't care about the precise formatting of the names, we could just check whether the name of the function is in the string (and change the name to something more unique): assertIn("my_first_step_target", step_in_targets[0]["label"])

Sorry for the late follow up, I got pulled into fixing something else. I think this is a simple enough solution. I'll reupload and reping for review.

**Comment by kendalharland** [comment]
> > I'm not sure if these names come from the demangler or debug info, but windows has different implementations for both, so it not surprising they're different. Assuming we don't care about the precise formatting of the names, we could just check whether the name of the function is in the string (and change the name to something more unique): assertIn("my_first_step_target", step_in_targets[0]["label"])
> 
> Sorry for the late follow up, I got pulled into fixing something else. I think this is a simple enough solution. I'll reupload and re-ping for review.

Updated in the latest commit.

**Comment by labath** [comment]
This looks fine to me. @jeffreytan81  ?

**Review by jeffreytan81 — APPROVED** [review]
Sorry, was in the middle of review then got interrupted/distracted by other stuff yesterday.

Looks good.

**Comment by kendalharland** [comment]
> Sorry, was in the middle of review then got interrupted/distracted by other stuff yesterday.
> 
> Looks good.

No problem at all. Thanks for the reviews! I'll look into the build failures.

**Comment by labath** [comment]
> AssertionError: 'funcB' not found in 'funcA()' : expect funcB

The step targets are coming in different order. It's probably an ABI thing, as the compiler produces the calls in different order as well: https://godbolt.org/z/cPqhsWba6

I guess we need to adjust the test to accept both orders.

**Comment by kendalharland** [comment]
Nice find! I didn't now about godbolt.org.  Updated the test to be order-independent w.r.t. funcA and funcB.

File: lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py [hunk 1]
```diff
@@ -55,14 +55,23 @@ def test_basic(self):
...
+            else:
+                self.fail(f"Unexpected step in target: {target}")
+
+        self.assertIsNotNone(funcA_target, "expect funcA")
+        self.assertIsNotNone(funcB_target, "expect funcB")
+        self.assertIn("foo", step_in_targets[2]["label"], "expect foo")
 
-        # Choose to step into second target and verify that we are in bar2()
-        self.stepIn(threadId=tid, targetId=step_in_targets[1]["id"], waitForStop=True)
+        # Choose to step into second target and verify that we are in funcB()
```
Comment (labath) [inline lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py:73]: funcB is not necessarily going to be the second target, so this comment is incorrect. However, instead of changing the comment, I think it'd be actually better to change the code so that it steps into the second function, regardless of what that function might be. This is because we would end up in the first function even if we did a "normal" step in operation rather than the fancy targetted one. Stepping into the second function verifies that we can correctly ignore the first function that gets called. (I wasn't the person who wrote the test, but I suspect this is roughly the motivation for it being written the way it is).

So basically, you could keep the `targetId=step_in_targets[1]["id"]` thing in the line below, but then change the expectation on line 77 to expect the function that we tried stepping into.

Comment (kendalharland) [reply lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py:73]: Thanks for the explanation! This sounds like the right thing to do, however I believe I'm hitting a bug: the test's DAP always steps into funcB regardless of the option passed to `targetId`.  Would you be able to reproduce this? If it reproduces for you, I wonder if this has always been broken, given that IIRC there's no Windows x64 CI and this test only runs when the architecture is x86_64.

Comment (labath) [reply lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py]: I don't have access to a windows machine right now, but I can totally believe it that this has never worked there. It's quite possible that there's some platform-specific bit needed for this to work, which simply hasn't been implemented for windows yet. I don't really know what that could be though, as I don't know how this feature works.

Is this something you'd be willing to look into? If yes, you can try debugging lldb to see where things start to break down. Otherwise, we can just xfail the test and move on.

Comment (jeffreytan81) [reply lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py]: I do not think anyone will have time to get a Windows machine, investigate and fix there. If there is indeed a bug on Windows, the easiest way is skip this test for Windows (with a comment mentioning it is not working there).

Comment (kendalharland) [reply lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py]: That's reasonable. I'll mark as disabled for now and file a bug to investigate. I can look into it once I get the time.

Comment (kendalharland) [reply lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py]: Updated in the latest commit.

Comment (kendalharland) [reply lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py]: I used @expectFailureAll rather than @skipIf, not realizing that the test _must_ fail if the latter is used. Updated in the latest commit. I'll need help landing as I don't have write access. Thank you all for the reviews!

File: lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py [hunk 2]
```diff
@@ -10,9 +10,10 @@
 
 
 class TestDAP_stepInTargets(lldbdap_testcase.DAPTestCaseBase):
-    @skipIf(
-        archs=no_match(["x86_64"])
-    )  # InstructionControlFlowKind for ARM is not supported yet.
+    @skipIf
+    # InstructionControlFlowKind for ARM is not supported yet.
+    # On x86_64 lldb-dap seems to ignore targetId when stepping into functions.
```
Comment (labath) [inline lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py:15]: Judging by everything said on this thread, I believe this is only true on x86_64 *windows*, so we should limit the exclusion to windows. Then you may actually be able to keep the xfail decorator, since the test is really expected to fail in this configuration (until the issue is fixed). (i.e., I think this should use `@expectedFailureAll(oslist=["windows"])` instead of the `@skipIf`).

Comment (kendalharland) [reply lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py]: Ah thanks for catching that. I added back the original `@skipIf` since the comment indicates that running the test on ARM is a waste, given that InstructionControlFlowKind is unsupported, and combined that with the `@expectedFailureAll` you gave above to expect failure on Windows configurations.

**Comment by kendalharland** [comment]
I'll need help merging from someone with write access

File: lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py (hunk 2 above)
Comment (labath) [reply lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py]: Great, that's exactly what I had in mind. Thanks.
